### PR TITLE
 Social badge support for static previews

### LINKED ISSUE #2479: Support static preview with social style
The social badges have typically included `?style=social`. However `staticPreview` returns `serviceData` which does not support the `style` parameter. This makes it impossible to render a static preview for the social badges.

e.g. 

https://github.com/badges/shields/blob/f8b0472b4c26a7bafcf4c91a527414054d5ac922/services/github/github-followers.service.js#L32-L36

This also came up in #2473 where I was adding an Endpoint badge, and thinking we should make it possible for a developer to make a badge which defaults to a particular style.

I'm not totally convinced we should do that. In some ways it violates a user's expectation, which is that Shields will always default to flat. Should we also let the developer specify `flat-square` or `for-the-badge`? At the same time, these are _custom_ badges, and it's useful to be able to include especially logos by default. Maybe it's okay, for custom badges.

### PR DESCRIPTION
**Pull request by paulmelnikow**
The static previews don't support the social badges. Adding that lets us remove support for `exampleUrl`. Close #2479.

This includes `style` and `namedLogo` in the service-definition export and updates the frontend to use it. To accomplish this, it passes `namedLogo` through `coalesceBadge`. After logo resolution is moved to `makeBadge` this duplication can be removed, as `logo` will no longer be needed in the result of `coalesceBadge`.

### CODE REVIEW

File: frontend/components/markup-modal/builder-common.js [hunk 1]
```diff
@@ -32,11 +32,12 @@ BuilderContainer.propTypes = {
 const labelFont = `
   font-family: system-ui;
   font-size: 11px;
-  text-transform: lowercase;
 `
 
 const BuilderLabel = styled.label`
   ${labelFont}
+
+  text-transform: lowercase;
```
Comment (paulmelnikow) [inline frontend/components/markup-modal/builder-common.js]: Don't lowercase capital letters in example text (e.g. **e.g. Follow**).

File: frontend/components/markup-modal/query-string-builder.js [hunk 2]
```diff
@@ -48,8 +48,9 @@ export default class QueryStringBuilder extends React.Component {
     })
 
     const badgeOptions = {}
-    supportedBadgeOptions.forEach(({ name, defaultValue = '' }) => {
-      badgeOptions[name] = defaultValue
+    const defaults = { style: defaultStyle }
```
Comment (paulmelnikow) [inline frontend/components/markup-modal/query-string-builder.js]: Allow the caller to dynamically set the default style.

**Comment by calebcartwright** [comment]
I haven't looked at most of these files before, but the changes here look pretty straightforward to me 👍

File: core/base-service/transform-example.js [hunk 3]
```diff
@@ -143,13 +144,21 @@ function transformExample(inExample, index, ServiceClass) {
...
       staticPreview,
       ServiceClass.defaultBadgeData,
       ServiceClass
     )
-    preview = { label, message: `${message}`, color }
+    preview = {
+      label,
+      message: `${message}`,
+      color,
+      style: style === 'flat' ? undefined : style,
```
[the remark this answers is not in the document]
Comment (paulmelnikow) [reply core/base-service/transform-example.js]: `flat` is the default style so it's what gets used for examples even if it's not specified.

**Pull request merged**